CSharp/Library/Microsoft.Bot.Builder/Dialogs/PromptDialog.cs

```diff
@@ -637,7 +637,7 @@ protected override bool TryParse(IMessageActivity message, out double result)
         [Serializable]
         public class PromptChoice<T> : Prompt<T, T>
         {
-            private IDictionary<T, IEnumerable<T>> choices;
+            private IDictionary<T, IEnumerable<T>> synonyms;
```

Comment:
This is a good point. `IDictionary<T, IEnumerable<T>> choices` can become part of PromptOptions. A new constructor can be added to PromptOptions that populate the choices. The old options-based constructor can populate the choices dictionary and the Options property return choices keys.